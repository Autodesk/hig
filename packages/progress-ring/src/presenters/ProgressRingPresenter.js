import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";

import SVGPresenter from "./SVGPresenter";

import Small from "./smallSVG";
import Medium from "./mediumSVG";
import ExtraSmall from "./xsmallSVG";
import { availableSizes, availableSurfaces } from "../constants";
import stylesheet from "./stylesheet";

const sizes = {
  xs: {
    svg: ExtraSmall,
    size: 20
  },
  s: {
    svg: Small,
    size: 28
  },
  m: {
    svg: Medium,
    size: 72
  },
  l: {
    svg: Medium,
    size: 144
  },
  xl: {
    svg: Medium,
    size: 242
  }
};

export default function ProgressRingPresenter(props) {
  const {
    innerRef,
    percentComplete,
    size,
    cssTransitionState,
    surface,
    mask
  } = props;
  const SVG = sizes[size].svg;
  const originalSize = sizes[size].svg === Medium ? 72 : sizes[size].size;

  return (
    <div
      className={css(stylesheet(props, {}).ring)}
      role="progressbar"
      aria-valuemin="0"
      aria-valuemax="100"
      aria-valuenow={percentComplete}
      ref={innerRef}
    >
      <SVGPresenter
        height={sizes[size].size}
        width={sizes[size].size}
        original={originalSize}
        svgData={SVG}
        cssTransitionState={cssTransitionState}
        surface={surface}
        mask={mask}
      />
    </div>
  );
}

ProgressRingPresenter.defaultProps = { size: "m", surface: 100 };

ProgressRingPresenter.propTypes = {
  innerRef: PropTypes.func,
  percentComplete: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  size: PropTypes.oneOf(availableSizes),
  cssTransitionState: PropTypes.string,
  surface: PropTypes.oneOf(availableSurfaces),
  mask: PropTypes.string
};
