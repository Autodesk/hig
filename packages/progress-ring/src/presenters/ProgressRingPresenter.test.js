import React from "react";
import renderer from "react-test-renderer";
import ProgressRingPresenter from "./ProgressRingPresenter";

describe("progress-ring/presenters/ProgressRingPresenter", () => {
  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        percentComplete: 33,
        size: "s"
      }
    },
    {
      description: "renders with surface prop",
      props: {
        percentComplete: 33,
        surface: 200
      }
    },
    {
      description: "renders with mask prop",
      props: {
        percentComplete: 33,
        mask: "red"
      }
    },
    {
      description: "renders with both surface & mask props",
      props: {
        percentComplete: 33,
        surface: 200,
        mask: "red"
      }
    }
  ].forEach(({ description, props: { ...otherProps } }) => {
    it(description, () => {
      const presenter = <ProgressRingPresenter {...otherProps} />;
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
