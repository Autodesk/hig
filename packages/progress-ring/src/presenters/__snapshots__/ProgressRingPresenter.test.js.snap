// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-ring/presenters/ProgressRingPresenter renders with all props 1`] = `
.emotion-18 {
  font-family: ArtifaktElement;
  line-height: 1.6;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: rgba(0,0,0,0.05);
}

.emotion-1 {
  opacity: 0;
}

.emotion-1:nth-of-type(even) {
  fill: #0686c2;
}

.emotion-1:nth-of-type(odd) {
  fill: #38abdf;
}

.emotion-17 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: #ffffff;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-18"
  role="progressbar"
>
  <svg
    height={28}
    version="1.1"
    viewBox="0 0 28 28"
    width={28}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <polygon
      className="hig__progress-ring__background emotion-0"
      fill={undefined}
      fillRule={undefined}
      points="22.49 5.51 14 2 5.51 5.51 2 14 5.51 22.49 14 26 22.49 22.49 26 14"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="5.52 5.52 14 6.33 14 2"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="5.52 5.52 8.58 8.58 14 6.33"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="2 14 8.58 8.58 5.52 5.52"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="2 14 6.33 14 8.58 8.58"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="5.52 22.48 6.33 14 2 14"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="5.52 22.48 8.58 19.42 6.33 14"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="14 26 8.58 19.42 5.52 22.48"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="14 26 14 21.67 8.58 19.42"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="22.48 22.48 14 21.67 14 26"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="22.48 22.48 19.42 19.42 14 21.67"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="26 14 19.42 19.42 22.48 22.48"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="26 14 21.67 14 19.42 19.42"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="22.48 5.52 21.67 14 26 14"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="22.48 5.52 19.42 8.58 21.67 14"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="14 2 19.42 8.58 22.48 5.52"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="14 2 14 6.33 19.42 8.58"
    />
    <polygon
      className="hig__progress-ring__mask emotion-17"
      fill="#FFFFFF"
      fillRule="nonzero"
      points="19.42 19.42 14 21.67 8.58 19.42 6.33 14 8.58 8.58 14 6.33 19.42 8.58 21.67 14"
    />
  </svg>
</div>
`;

exports[`progress-ring/presenters/ProgressRingPresenter renders with both surface & mask props 1`] = `
.emotion-22 {
  font-family: ArtifaktElement;
  line-height: 1.6;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: rgba(0,0,0,0.05);
}

.emotion-1 {
  opacity: 0;
}

.emotion-1:nth-of-type(even) {
  fill: #0686c2;
}

.emotion-1:nth-of-type(odd) {
  fill: #38abdf;
}

.emotion-21 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: red;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-22"
  role="progressbar"
>
  <svg
    height={72}
    version="1.1"
    viewBox="0 0 72 72"
    width={72}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <polygon
      className="hig__progress-ring__background emotion-0"
      fill={undefined}
      fillRule={undefined}
      points="54.81 10.11 36 4 17.19 10.11 5.57 26.11 5.57 45.89 17.19 61.89 36 68 54.81 61.89 66.43 45.89 66.43 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 10.31 36 4 17.19 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 10.31 17.19 10.11 20.9 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 15.215 17.19 10.11 5.565 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 15.215 5.565 26.11 11.565 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 28.06 5.565 26.11 5.565 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 28.06 5.565 45.89 11.565 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 43.94 5.565 45.89 17.19 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 43.94 17.19 61.89 20.9 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 56.785 17.19 61.89 36 68"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 56.785 36 68 36 61.69"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 61.69 36 68 54.81 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 61.69 54.81 61.89 51.1 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 56.785 54.81 61.89 66.435 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 56.785 66.435 45.89 60.435 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 43.94 66.435 45.89 66.435 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 43.94 66.435 26.11 60.435 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 28.06 66.435 26.11 54.81 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 28.06 54.81 10.11 51.1 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 15.215 54.81 10.11 36 4"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 15.215 36 4 36 10.31"
    />
    <polygon
      className="hig__progress-ring__mask emotion-21"
      fill="#FFFFFF"
      fillRule="nonzero"
      points="60.43 43.94 51.1 56.78 36 61.69 20.9 56.78 11.57 43.94 11.57 28.06 20.9 15.22 36 10.31 51.1 15.22 60.43 28.07"
    />
  </svg>
</div>
`;

exports[`progress-ring/presenters/ProgressRingPresenter renders with mask prop 1`] = `
.emotion-22 {
  font-family: ArtifaktElement;
  line-height: 1.6;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: rgba(0,0,0,0.05);
}

.emotion-1 {
  opacity: 0;
}

.emotion-1:nth-of-type(even) {
  fill: #0686c2;
}

.emotion-1:nth-of-type(odd) {
  fill: #38abdf;
}

.emotion-21 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: red;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-22"
  role="progressbar"
>
  <svg
    height={72}
    version="1.1"
    viewBox="0 0 72 72"
    width={72}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <polygon
      className="hig__progress-ring__background emotion-0"
      fill={undefined}
      fillRule={undefined}
      points="54.81 10.11 36 4 17.19 10.11 5.57 26.11 5.57 45.89 17.19 61.89 36 68 54.81 61.89 66.43 45.89 66.43 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 10.31 36 4 17.19 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 10.31 17.19 10.11 20.9 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 15.215 17.19 10.11 5.565 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 15.215 5.565 26.11 11.565 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 28.06 5.565 26.11 5.565 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 28.06 5.565 45.89 11.565 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 43.94 5.565 45.89 17.19 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 43.94 17.19 61.89 20.9 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 56.785 17.19 61.89 36 68"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 56.785 36 68 36 61.69"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 61.69 36 68 54.81 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 61.69 54.81 61.89 51.1 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 56.785 54.81 61.89 66.435 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 56.785 66.435 45.89 60.435 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 43.94 66.435 45.89 66.435 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 43.94 66.435 26.11 60.435 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 28.06 66.435 26.11 54.81 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 28.06 54.81 10.11 51.1 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 15.215 54.81 10.11 36 4"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 15.215 36 4 36 10.31"
    />
    <polygon
      className="hig__progress-ring__mask emotion-21"
      fill="#FFFFFF"
      fillRule="nonzero"
      points="60.43 43.94 51.1 56.78 36 61.69 20.9 56.78 11.57 43.94 11.57 28.06 20.9 15.22 36 10.31 51.1 15.22 60.43 28.07"
    />
  </svg>
</div>
`;

exports[`progress-ring/presenters/ProgressRingPresenter renders with surface prop 1`] = `
.emotion-22 {
  font-family: ArtifaktElement;
  line-height: 1.6;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: rgba(0,0,0,0.05);
}

.emotion-1 {
  opacity: 0;
}

.emotion-1:nth-of-type(even) {
  fill: #0686c2;
}

.emotion-1:nth-of-type(odd) {
  fill: #38abdf;
}

.emotion-21 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: #f5f5f5;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-22"
  role="progressbar"
>
  <svg
    height={72}
    version="1.1"
    viewBox="0 0 72 72"
    width={72}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <polygon
      className="hig__progress-ring__background emotion-0"
      fill={undefined}
      fillRule={undefined}
      points="54.81 10.11 36 4 17.19 10.11 5.57 26.11 5.57 45.89 17.19 61.89 36 68 54.81 61.89 66.43 45.89 66.43 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 10.31 36 4 17.19 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 10.31 17.19 10.11 20.9 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 15.215 17.19 10.11 5.565 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 15.215 5.565 26.11 11.565 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 28.06 5.565 26.11 5.565 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 28.06 5.565 45.89 11.565 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 43.94 5.565 45.89 17.19 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 43.94 17.19 61.89 20.9 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 56.785 17.19 61.89 36 68"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 56.785 36 68 36 61.69"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 61.69 36 68 54.81 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 61.69 54.81 61.89 51.1 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 56.785 54.81 61.89 66.435 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 56.785 66.435 45.89 60.435 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 43.94 66.435 45.89 66.435 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 43.94 66.435 26.11 60.435 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 28.06 66.435 26.11 54.81 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 28.06 54.81 10.11 51.1 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 15.215 54.81 10.11 36 4"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 15.215 36 4 36 10.31"
    />
    <polygon
      className="hig__progress-ring__mask emotion-21"
      fill="#FFFFFF"
      fillRule="nonzero"
      points="60.43 43.94 51.1 56.78 36 61.69 20.9 56.78 11.57 43.94 11.57 28.06 20.9 15.22 36 10.31 51.1 15.22 60.43 28.07"
    />
  </svg>
</div>
`;

exports[`progress-ring/presenters/ProgressRingPresenter renders without props 1`] = `
.emotion-22 {
  font-family: ArtifaktElement;
  line-height: 1.6;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: rgba(0,0,0,0.05);
}

.emotion-1 {
  opacity: 0;
}

.emotion-1:nth-of-type(even) {
  fill: #0686c2;
}

.emotion-1:nth-of-type(odd) {
  fill: #38abdf;
}

.emotion-21 {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  fill: #ffffff;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={undefined}
  className="emotion-22"
  role="progressbar"
>
  <svg
    height={72}
    version="1.1"
    viewBox="0 0 72 72"
    width={72}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <polygon
      className="hig__progress-ring__background emotion-0"
      fill={undefined}
      fillRule={undefined}
      points="54.81 10.11 36 4 17.19 10.11 5.57 26.11 5.57 45.89 17.19 61.89 36 68 54.81 61.89 66.43 45.89 66.43 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 10.31 36 4 17.19 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 10.31 17.19 10.11 20.9 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 15.215 17.19 10.11 5.565 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 15.215 5.565 26.11 11.565 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 28.06 5.565 26.11 5.565 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 28.06 5.565 45.89 11.565 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="11.565 43.94 5.565 45.89 17.19 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="11.565 43.94 17.19 61.89 20.9 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="20.9 56.785 17.19 61.89 36 68"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="20.9 56.785 36 68 36 61.69"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="36 61.69 36 68 54.81 61.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="36 61.69 54.81 61.89 51.1 56.785"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 56.785 54.81 61.89 66.435 45.89"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 56.785 66.435 45.89 60.435 43.94"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 43.94 66.435 45.89 66.435 26.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 43.94 66.435 26.11 60.435 28.06"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="60.435 28.06 66.435 26.11 54.81 10.11"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="60.435 28.06 54.81 10.11 51.1 15.215"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0ED3BE"
      fillRule="nonzero"
      points="51.1 15.215 54.81 10.11 36 4"
    />
    <polygon
      className="hig__progress-ring__segment emotion-1"
      fill="#0CA899"
      fillRule="nonzero"
      points="51.1 15.215 36 4 36 10.31"
    />
    <polygon
      className="hig__progress-ring__mask emotion-21"
      fill="#FFFFFF"
      fillRule="nonzero"
      points="60.43 43.94 51.1 56.78 36 61.69 20.9 56.78 11.57 43.94 11.57 28.06 20.9 15.22 36 10.31 51.1 15.22 60.43 28.07"
    />
  </svg>
</div>
`;
